src/intrinsics/prepack/global.js

```diff
@@ -227,14 +227,19 @@ export default function(realm: Realm): void {
       "__evaluatePureFunction",
       0,
       (context, [functionValue]) => {
-        invariant(functionValue instanceof ECMAScriptSourceFunctionValue);
-        invariant(typeof functionValue.$Call === "function");
-        let functionCall: Function = functionValue.$Call;
-        return realm.evaluatePure(
-          () => functionCall(realm.intrinsics.undefined, []),
-          /*bubbles*/ true,
-          /*reportSideEffectFunc*/ null
-        );
+        if (functionValue instanceof ECMAScriptSourceFunctionValue && typeof functionValue.$Call === "function") {
+          let functionCall: Function = functionValue.$Call;
+          return realm.evaluatePure(
+            () => functionCall(realm.intrinsics.undefined, []),
+            /*bubbles*/ true,
+            /*reportSideEffectFunc*/ null
+          );
+        } else {
+          realm.handleError(
+            new CompilerDiagnostic(`CompilerDiagnostic Error`, realm.currentLocation, "PP1005", "Warning")
```

Comment:
Since this is a warning, it is recoverable and you should throw the fatal error only if the return result of handleError is "Fail".

Reply:
I believe it should not be a warning but an error as bug says 

'In the implementation, it shouldn't check via an invariant that the argument is a proper value, but instead, it should throw a user-level exception or emit a CompilerDiagnostics error.'

Should I replace warning with error?

Reply:
In that case then yes, "FatalError" not "Warning". And then don't recover.